bot/cogs/translate_cog.py

```diff
@@ -0,0 +1,179 @@
+            await self.translate_given_lang(ctx, input)
+        else:
+            await self.translate_detect_lang(ctx, input)
+    
+    async def translate_given_lang(self, ctx, input):
+        input_lang = input[0]
+        output_lang = input[1]
+        text = ''
+        for i in input[2:]:
+            text += f'{i} '
+
+        params = {}
+
+        log.info('Input Lang Code: ' + str(get_lang_code(input_lang)))
+        log.info('Output Lang Code: ' + str(get_lang_code(output_lang)))
+        params = {
+            'api-version': '3.0',
+            'from': get_lang_code(input_lang),
+            'to': get_lang_code(output_lang)
+        }
+       
+        body = [{
+            'text': text
+        }]
+
+        async with aiohttp.ClientSession() as session:
+            async with await session.post(url = TRANSLATE_API_URL, params = params, headers = headers, json = body) as resp: 
+                    response = json.loads(await resp.text())
+
+        log.info(response[0]['translations'])
+        embed = discord.Embed(title='Translate', color = Colors.ClemsonOrange)
+        name = 'Translated to ' + language_short_code_to_name[response[0]['translations'][0]['to']]
+        embed.add_field(name=name, value = response[0]['translations'][0]['text'], inline=False)
+        await ctx.send(embed=embed)
+        return
+        
+    async def translate_detect_lang(self, ctx, input):
+        output_lang = input[0]
+        text = ''
+        for i in input[1:]:
+            text += f'{i} '
+
+        log.info('Output Lang Code: ' + str(get_lang_code(output_lang)))
+        
+        params = {
+            'api-version': '3.0',
+            'to': get_lang_code(output_lang)
+        }
+        
+        body = [{
+            'text': text
+        }]
+        
+        async with aiohttp.ClientSession() as session:
+            async with await session.post(url = TRANSLATE_API_URL, params = params, headers = headers, json = body) as resp: 
+                    response = json.loads(await resp.text())
+        
+        log.info(response[0]['detectedLanguage'])
+        log.info(response[0]['translations'])
+        embed = discord.Embed(title='Translate', color = Colors.ClemsonOrange)
+        name = 'Translated to ' + language_short_code_to_name[response[0]['translations'][0]['to']]
+        embed.add_field(name=name, value = response[0]['translations'][0]['text'], inline=False)
+        embed.add_field(name='Confidence Level:', value = response[0]['detectedLanguage']['score'], inline=True)
+        embed.add_field(name='Detected Language:', value = language_short_code_to_name[response[0]['detectedLanguage']['language']], inline=True)
+        await ctx.send(embed=embed)
+        return
+
+def is_valid_lang_code(input: str):
+    return input.lower() in language_short_code_to_name or input.lower() in language_name_to_short_code
+
+def get_lang_code(input: str):
+    if input.lower() in language_short_code_to_name:
```

Comment:
you should merge these dicts so finding a code isnt so convoluted

Reply:
So the reason they are separate is because regardless of if 'english' or 'en' is given, i have to pass 'en' to microsoft. If i have long form -> short form and short form -> long form in one dictionary, it makes things harder to get the short form